### DISCUSSION TITLE
Compatibility issue with return type in `newCollection` method

### DISCUSSION THREAD

**Discussion by nbayramberdiyev** [larastan/larastan#1996] [Q&A] [answered]
After upgrading Laravel to v11.15.0, I encountered the following error:

> Return type (App\Models\Collections\Regions) of method App\Models\Region::newCollection() should be compatible with return type (Illuminate\Database\Eloquent\Collection<(int|string), Illuminate\Database\Eloquent\Model>) of method Illuminate\Database\Eloquent\Model::newCollection()

Here's the `Region` model:
```php
namespace App\Models;

use App\Models\Collections\Regions;
use Illuminate\Database\Eloquent\Model;

final class Region extends Model
{
    // ...

    /**
     * @param array<int, self> $models
     */
    public function newCollection(array $models = []): Regions
    {
        return new Regions($models);
    }
}
```

And the `Regions` collection:
```php
namespace App\Models\Collections;

use App\Models\Region;
use Illuminate\Database\Eloquent\Collection;

/**
 * @extends Collection<int, Region>
 */
final class Regions extends Collection
{
    // ...
}
```

Additionally, my `phpstan.neon` file looks like this:
```neon
includes:
    - ./vendor/larastan/larastan/extension.neon

parameters:
    ignoreErrors:
        - identifier: missingType.generics

    paths:
        - app
        - config
        - database
        - tests

    level: 8
```

Laravel: v11.15.0
Larastan: v2.9.8

What could be causing this issue?

**Comment by calebdw** [comment]
This looks like a PHPStan issue: https://github.com/phpstan/phpstan/issues/11354

**Reply by calebdw** [reply, marked as answer]
I've opened a PR at the framwork to fix: https://github.com/laravel/framework/pull/52171

**Reply by nbayramberdiyev** [reply]
Thanks!

**Reply by szepeviktor** [reply]
Merged.
